### PR TITLE
core: add health command in go

### PR DESCRIPTION
**Pull request by subhamkrai**
core: add health command in go

Signed-off-by: subhamkrai <srai@redhat.com>

These changes are on top of the mon quorum pr, so check the last commit only. Thanks!

command
`kubectl rook-ceph health`

output
```
Info:  Checking if at least three mon pods are running on different nodes

Warning: At least three mon pods should running on different nodesrook-ceph-mon-a-655fbb547-7fbbn        Running         rook-ceph       minikube
rook-ceph-mon-b-6f74686789-bl5lj         Running         rook-ceph       minikube
rook-ceph-mon-c-7676c5485-bbg2v          Running         rook-ceph       minikube

Info:  Checking mon quorum and ceph health details


Info:  Checking if at least three osd pods are running on different nodes

Warning: At least three osd pods should running on different nodesrook-ceph-osd-0-599db9876f-qzv7f       Running         rook-ceph       minikube
rook-ceph-osd-1-77959f9db5-j2nw9         Running         rook-ceph       minikube

Info: Pods that are in 'Running' status
csi-cephfsplugin-provisioner-84cc595b78-m6xvt    Running         rook-ceph       minikube
csi-cephfsplugin-sgrx5   Running         rook-ceph       minikube
csi-rbdplugin-khg6h      Running         rook-ceph       minikube
csi-rbdplugin-provisioner-6f6b6b8cd6-s6jss       Running         rook-ceph       minikube
rook-ceph-mgr-a-5dd87bf686-xnkwp         Running         rook-ceph       minikube
rook-ceph-mon-a-655fbb547-7fbbn          Running         rook-ceph       minikube
rook-ceph-mon-b-6f74686789-bl5lj         Running         rook-ceph       minikube
rook-ceph-mon-c-7676c5485-bbg2v          Running         rook-ceph       minikube
rook-ceph-operator-cf4f7dfd4-4845g       Running         rook-ceph       minikube
rook-ceph-osd-0-599db9876f-qzv7f         Running         rook-ceph       minikube
rook-ceph-osd-1-77959f9db5-j2nw9         Running         rook-ceph       minikube
csi-cephfsplugin-provisioner-84cc595b78-m6xvt    Running         rook-ceph       minikube
csi-cephfsplugin-sgrx5   Running         rook-ceph       minikube
csi-rbdplugin-khg6h      Running         rook-ceph       minikube
csi-rbdplugin-provisioner-6f6b6b8cd6-s6jss       Running         rook-ceph       minikube
rook-ceph-mgr-a-5dd87bf686-xnkwp         Running         rook-ceph       minikube
rook-ceph-mon-a-655fbb547-7fbbn          Running         rook-ceph       minikube
rook-ceph-mon-b-6f74686789-bl5lj         Running         rook-ceph       minikube
rook-ceph-mon-c-7676c5485-bbg2v          Running         rook-ceph       minikube
rook-ceph-operator-cf4f7dfd4-4845g       Running         rook-ceph       minikube
rook-ceph-osd-0-599db9876f-qzv7f         Running         rook-ceph       minikube
rook-ceph-osd-1-77959f9db5-j2nw9         Running         rook-ceph       minikube

Warning: Pods that are 'Not' in 'Running' status
rook-ceph-osd-prepare-minikube-m6dks     Succeeded       rook-ceph       minikube
rook-ceph-osd-prepare-minikube-m6dks     Succeeded       rook-ceph       minikube
Info:  checking placement group status


INFO: 32 pgs: 32 active+clean; 463 KiB data, 40 MiB used, 39 GiB / 39 GiB avail


Info:  checking if at least one mgr pod is running
rook-ceph-mgr-a-5dd87bf686-xnkwp         Running         rook-ceph       minikube
```

### CODE REVIEW

**Review by subhamkrai — COMMENTED** [review]
The logs are not mirrored images of what we have in bash, since keeping the same logs as bash is difficult to maintain in golang. Also, will have a pr in the future(very soon to modify the logging system)

File: pkg/health/health.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+func checkMgrPodsStatusAndCounts(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	opts := metav1.ListOptions{LabelSelector: "app=rook-ceph-mgr"}
+	podList, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), opts)
+	if err != nil {
+		fmt.Printf("\nfailed to list mgr pods with label %s: %v\n", opts.LabelSelector, err)
+	}
+
+	if len(podList.Items) < 1 {
+		fmt.Println("At least one mgr pod should running")
```
Comment (travisn) [inline pkg/health/health.go:156]:
```suggestion
		fmt.Println("At least one mgr pod should be running")
```

File: pkg/health/health.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	if err != nil {
+		fmt.Printf("\nfailed to list pods in namespace %s: %v\n", operatorNamespace, err)
+	}
+
+	podListInClNs, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("\nfailed to list pods in namespace %s: %v\n", clusterNamespace, err)
+	}
+
+	var PodNotRunning, PodRunning []v1.Pod
```
Comment (travisn) [inline pkg/health/health.go:107]: Local var names should be lower case
```suggestion
	var podNotRunning, podRunning []v1.Pod
```

File: pkg/health/health.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+	}
+}
+
+func CheckAllPodsStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	podListInOpNs, err := context.Clientset.CoreV1().Pods(operatorNamespace).List(ctx.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("\nfailed to list pods in namespace %s: %v\n", operatorNamespace, err)
+	}
+
+	podListInClNs, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), metav1.ListOptions{})
```
Comment (travisn) [inline pkg/health/health.go:102]:
```suggestion
	clusterNamespacePods, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), metav1.ListOptions{})
```

File: pkg/health/health.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+		fmt.Printf("\n%s", cephHealthDetails)
+	} else if cephHealthDetails == "HEALTH_WARN" {
+		fmt.Println("Warning: ", cephHealthDetails)
+	} else if cephHealthDetails == "HEALTH_ERR" {
+		fmt.Println("Error: ", cephHealthDetails)
+	}
+}
+
+func CheckAllPodsStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	podListInOpNs, err := context.Clientset.CoreV1().Pods(operatorNamespace).List(ctx.TODO(), metav1.ListOptions{})
```
Comment (travisn) [inline pkg/health/health.go:97]: for better readability, spell out var names. If they're long, then find simpler names.
```suggestion
	operatorNamespacePods, err := context.Clientset.CoreV1().Pods(operatorNamespace).List(ctx.TODO(), metav1.ListOptions{})
```

File: pkg/health/health.go [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+	}
+
+	for i := range podList.Items {
+		fmt.Println(podList.Items[i].Name, "\t", podList.Items[i].Status.Phase, "\t", podList.Items[i].Namespace, "\t", podList.Items[i].Spec.NodeName)
+	}
+}
+
+func checkMonQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	cephHealthDetails := exec.RunCommandInOperatorPod(context, "ceph", []string{"health", "detail"}, operatorNamespace, clusterNamespace, false)
+	if cephHealthDetails == "HEALTH_OK" {
```
Comment (travisn) [inline pkg/health/health.go:87]: `cephHealthDetails` contains a lot of status, right? Seems like this will never be equal.

File: pkg/health/health.go [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+	fmt.Println("Info:  checking placement group status")
+	checkPgStatus(context, operatorNamespace, clusterNamespace, exitOnError)
+
+	fmt.Println()
+	fmt.Println("Info:  checking if at least one mgr pod is running")
+	checkMgrPodsStatusAndCounts(context, operatorNamespace, clusterNamespace, exitOnError)
+}
+
+func checkPodsOnNodes(context *k8sutil.Context, operatorNamespace, clusterNamespace, label string, exitOnError bool) {
```
Comment (travisn) [inline pkg/health/health.go:54]: not used
```suggestion
func checkPodsOnNodes(context *k8sutil.Context, operatorNamespace, clusterNamespace, label string) {
```

File: pkg/health/health.go [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+)
+
+func Health(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
```
Comment (travisn) [inline pkg/health/health.go:31]: No need for the parameter, just define the local var to keep the method signature simpler. Or maybe this local var isn't even needed if each of these child methods can just pass `false`
```suggestion
func Health(context *k8sutil.Context, operatorNamespace, clusterNamespace string) {
   exitOnError := true
```

**Review by travisn — CHANGES_REQUESTED** [review]
mainly some nits

File: pkg/health/health.go (hunk 2 above)
Comment (subhamkrai) [reply pkg/health/health.go]: one of my favourite mistakes 😅

File: pkg/health/health.go (hunk 5 above)
Comment (subhamkrai) [reply pkg/health/health.go]: right, currently only using `strings.Contains()` but I'm planning on working locally, to use json marshaling whenever we are comparing ceph output status but will not be part of this pr since it will require changes in other commands too.

File: pkg/health/health.go (hunk 7 above)
Comment (subhamkrai) [reply pkg/health/health.go]: got it

File: .github/workflows/go-test.yaml [hunk 8]
```diff
@@ -35,6 +35,8 @@ jobs:
       - name: Test plugin with ceph commands
         run: |
           set -e
+          kubectl rook-ceph health
+
           kubectl rook-ceph ceph status
```
Comment (travisn) [inline .github/workflows/go-test.yaml:40]:
```suggestion
      - name: ceph status
          kubectl rook-ceph ceph status
```

File: .github/workflows/go-test.yaml [hunk 9]
```diff
@@ -35,6 +35,8 @@ jobs:
       - name: Test plugin with ceph commands
```
Comment (travisn) [inline .github/workflows/go-test.yaml:35]: Can we get these tests all under different `name`s in the test? Or is that going to be a separate PR?
```suggestion
      - name: rook-ceph health
```

File: .github/workflows/go-test.yaml (hunk 8 above)
Comment (travisn) [reply .github/workflows/go-test.yaml:40]: and so on

File: pkg/health/health.go [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+)
+
+func Health(context *k8sutil.Context, operatorNamespace, clusterNamespace string) {
+	exitOnError := true
+
+	fmt.Println("Info:  Checking if at least three mon pods are running on different nodes")
```
Comment (travisn) [inline pkg/health/health.go:34]: Why the extra space?
```suggestion
	fmt.Println("Info: Checking if at least three mon pods are running on different nodes")
```

File: pkg/health/health.go [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+	checkMonQuorum(context, operatorNamespace, clusterNamespace, exitOnError)
+
+	fmt.Println()
+	fmt.Println("Info:  Checking if at least three osd pods are running on different nodes")
+	checkPodsOnNodes(context, operatorNamespace, clusterNamespace, "app=rook-ceph-osd")
+
+	fmt.Println()
+	CheckAllPodsStatus(context, operatorNamespace, clusterNamespace, exitOnError)
+
+	fmt.Println("Info:  checking placement group status")
```
Comment (travisn) [inline pkg/health/health.go:48]:
```suggestion
	fmt.Println("Info: Checking placement group status")
```

File: pkg/health/health.go [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+	checkPgStatus(context, operatorNamespace, clusterNamespace, exitOnError)
+
+	fmt.Println()
+	fmt.Println("Info:  checking if at least one mgr pod is running")
```
Comment (travisn) [inline pkg/health/health.go:52]:
```suggestion
	fmt.Println("Info: Checking if at least one mgr pod is running")
```

File: pkg/health/health.go [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+	if strings.Contains(cephHealthDetails, "HEALTH_OK") {
+		fmt.Printf("\n%s", cephHealthDetails)
+	} else if strings.Contains(cephHealthDetails, "HEALTH_WARN") {
+		fmt.Println("Warning: ", cephHealthDetails)
+	} else if strings.Contains(cephHealthDetails, "HEALTH_ERR") {
+		fmt.Println("Error: ", cephHealthDetails)
+	}
+}
+
+func CheckAllPodsStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
```
Comment (travisn) [inline pkg/health/health.go:98]: not used
```suggestion
func CheckAllPodsStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string) {
```

File: pkg/health/health.go [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+	operatorNamespacePods, err := context.Clientset.CoreV1().Pods(operatorNamespace).List(ctx.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("\nfailed to list pods in namespace %s: %v\n", operatorNamespace, err)
+	}
+
+	clusterNamespacePods, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), metav1.ListOptions{})
```
Comment (travisn) [inline pkg/health/health.go:104]: If the operator and cluster namespace are the same, will these pods be duplicate of the operatorNamespacePods?

File: pkg/health/health.go [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+	if err != nil {
+		fmt.Printf("\nfailed to list pods in namespace %s: %v\n", clusterNamespace, err)
+	}
+
+	var podNotRunning, podRunning []v1.Pod
+
+	for i := range operatorNamespacePods.Items {
```
Comment (travisn) [inline pkg/health/health.go:111]: How about a helper method for this `for` loop? so we don't have to duplicate it on line 119.

File: pkg/health/health.go [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+		fmt.Println(podRunning[i].Name, "\t", podRunning[i].Status.Phase, "\t", podRunning[i].Namespace, "\t", podRunning[i].Spec.NodeName)
+	}
+
+	fmt.Println("\nWarning: Pods that are 'Not' in 'Running' status")
+	for i := range podNotRunning {
+		fmt.Println(podNotRunning[i].Name, "\t", podNotRunning[i].Status.Phase, "\t", podNotRunning[i].Namespace, "\t", podNotRunning[i].Spec.NodeName)
+	}
+}
+
+func checkPgStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
```
Comment (travisn) [inline pkg/health/health.go:138]:
```suggestion
func checkPgStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string) {
```

File: pkg/health/health.go [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+	if strings.Contains(pg_stat_code, "active+clean;") {
+		fmt.Printf("\nINFO: %s\n", pg_stat_code)
+	} else if strings.Contains(pg_stat_code, "down") || strings.Contains(pg_stat_code, "incomplete") || strings.Contains(pg_stat_code, "snaptrim_error") {
+		fmt.Printf("\nError: %s\n", pg_stat_code)
+	} else {
+		fmt.Printf("\nWarning: %s\n", pg_stat_code)
+	}
+}
+
+func checkMgrPodsStatusAndCounts(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
```
Comment (travisn) [inline pkg/health/health.go:150]:
```suggestion
func checkMgrPodsStatusAndCounts(context *k8sutil.Context, operatorNamespace, clusterNamespace string) {
```

**Review by travisn — CHANGES_REQUESTED** [review]

File: pkg/health/health.go [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+	fmt.Println("\nWarning: Pods that are 'Not' in 'Running' status")
+	for i := range podNotRunning {
+		fmt.Println(podNotRunning[i].Name, "\t", podNotRunning[i].Status.Phase, "\t", podNotRunning[i].Namespace, "\t", podNotRunning[i].Spec.NodeName)
+	}
+}
+
+func checkPgStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	pg_stat_code := exec.RunCommandInOperatorPod(context, "ceph", []string{"pg", "stat"}, operatorNamespace, clusterNamespace, false)
+
+	if strings.Contains(pg_stat_code, "active+clean;") {
```
[the remark this answers is not in the document]
Comment (subhamkrai) [reply pkg/health/health.go:141]: right, we need to find a better way. 
We can try using the json parsing with ceph commands but n the other hand, it will put them on either `Info, Warn or Error` mode. We'll have to log for each pg status type.

Or We can just print the ceph command.
thoughts?

For, now I have move the condition checking for `error`  first so that is any of the error state is there is the pg status we say it is `Error`

File: pkg/health/health.go (hunk 15 above)
Comment (subhamkrai) [reply pkg/health/health.go:111]: done

File: pkg/health/health.go (hunk 14 above)
Comment (subhamkrai) [reply pkg/health/health.go:104]: good catch, now it should be good

File: pkg/health/health.go [hunk 19]
```diff
@@ -0,0 +1,161 @@
...
+	if len(nodeList) < 3 {
+		fmt.Printf("\nWarning: At least three %s pods should running on different nodes", daemonType)
+	}
+
+	for i := range podList.Items {
+		fmt.Println(podList.Items[i].Name, "\t", podList.Items[i].Status.Phase, "\t", podList.Items[i].Namespace, "\t", podList.Items[i].Spec.NodeName)
+	}
+}
+
+func checkMonQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
```
Comment (travisn) [inline pkg/health/health.go:87]: not used
```suggestion
func checkMonQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace string) {
```

File: pkg/health/health.go [hunk 20]
```diff
@@ -0,0 +1,161 @@
...
+	var nodeList = make(map[string]string)
+	for i := range podList.Items {
+		nodeName := podList.Items[i].Spec.NodeName
+		if _, okay := nodeList[nodeName]; !okay {
+			nodeList[nodeName] = podList.Items[i].Name
+		}
+	}
+
+	if len(nodeList) < 3 {
+		fmt.Printf("\nWarning: At least three %s pods should running on different nodes", daemonType)
```
Comment (travisn) [inline pkg/health/health.go:79]: Looks like the println on line 83 will end up on the same line as this warning
```suggestion
		fmt.Printf("\nWarning: At least three %s pods should running on different nodes\n", daemonType)
```

File: pkg/health/health.go [hunk 21]
```diff
@@ -0,0 +1,161 @@
...
+	}
+	return podRunning, podNotRunning
+}
+
+func checkPgStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string) {
+	pg_stat_code := exec.RunCommandInOperatorPod(context, "ceph", []string{"pg", "stat"}, operatorNamespace, clusterNamespace, false)
+
+	if strings.Contains(pg_stat_code, "down") || strings.Contains(pg_stat_code, "incomplete") || strings.Contains(pg_stat_code, "snaptrim_error") {
+		fmt.Printf("\nError: %s\n", pg_stat_code)
+	} else if strings.Contains(pg_stat_code, "active+clean;") {
```
Comment (travisn) [inline pkg/health/health.go:141]: There are a lot of pg statuses, I wonder if we are missing something. Can we check if the only status is exactly `active+clean` for this case?

File: pkg/health/health.go [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
+		fmt.Printf("\nInfo: %s\n", pg_stat_code)
+	} else {
+		fmt.Printf("\nWarning: %s\n", pg_stat_code)
```
Comment (travisn) [inline pkg/health/health.go:144]: I wonder if we can just print info/warning/error depending on ceph's HEALTH_OK, HEALTH_WARN, or HEALTH_ERR. Then we don't need to try and guess what the state is, we can just let ceph tell us. For the case where the health issue is because of something other than PGs, we could just print info if all the PGs are exactly `active+clean`, otherwise just match ceph's health status for this output.

File: pkg/health/health.go [hunk 23]
```diff
@@ -0,0 +1,161 @@
...
+	}
+}
+
+func checkMgrPodsStatusAndCounts(context *k8sutil.Context, operatorNamespace, clusterNamespace string) {
+	opts := metav1.ListOptions{LabelSelector: "app=rook-ceph-mgr"}
+	podList, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), opts)
+	if err != nil {
+		fmt.Printf("\nfailed to list mgr pods with label %s: %v\n", opts.LabelSelector, err)
```
Comment (travisn) [inline pkg/health/health.go:152]: need to return here? the `podList` won't be valid below
```suggestion
		fmt.Printf("\nfailed to list mgr pods with label %s: %v\n", opts.LabelSelector, err)
		return
```

**Review by travisn — CHANGES_REQUESTED** [review]

File: pkg/health/health.go (hunk 21 above)
Comment (subhamkrai) [reply pkg/health/health.go]: should be better now with Json parsing.

File: pkg/health/health.go (hunk 22 above)
Comment (subhamkrai) [reply pkg/health/health.go]: should be better now with Json parsing.

File: pkg/health/health.go (hunk 5 above)
Comment (travisn) [reply pkg/health/health.go:109]: I see, `unMarshalCephCommands()` only returns the status. Then we need to rename that method and also `cephHealthDetails` so it makes sense that it's only the status.

File: pkg/health/health.go [hunk 24]
```diff
@@ -0,0 +1,199 @@
...
+		fmt.Printf("\nWarning: At least three %s pods should running on different nodes\n", daemonType)
+	}
+
+	for i := range podList.Items {
+		fmt.Println(podList.Items[i].Name, "\t", podList.Items[i].Status.Phase, "\t", podList.Items[i].Namespace, "\t", podList.Items[i].Spec.NodeName)
+	}
+}
+
+func checkMonQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace string) {
+	cephHealthDetails, _ := unMarshalCephCommands(context, operatorNamespace, clusterNamespace)
```
Comment (travisn) [inline pkg/health/health.go:108]:
```suggestion
	cephHealthStatus, _ := unMarshalCephStatus(context, operatorNamespace, clusterNamespace)
```

**Review by travisn — CHANGES_REQUESTED** [review]

File: pkg/health/health.go (hunk 24 above)
Comment (subhamkrai) [reply pkg/health/health.go]: done

File: pkg/health/health.go [hunk 25]
```diff
@@ -0,0 +1,199 @@
...
+	if strings.Contains(label, "osd") {
+		daemonType = "osd"
+	} else if strings.Contains(label, "mon") {
+		daemonType = "mon"
+	}
+
+	opts := metav1.ListOptions{LabelSelector: label}
+	podList, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), opts)
+	if err != nil {
+		fmt.Printf("\nfailed to list %s pods with label %s: %v\n", daemonType, opts.LabelSelector, err)
```
Comment (travisn) [inline pkg/health/health.go:87]: return on error? The `podList` isn't valid.

**Review by travisn — APPROVED** [review]
Just one more small question, feel free to merge after fixing.

File: pkg/health/health.go (hunk 25 above)
Comment (subhamkrai) [reply pkg/health/health.go:87]: right, done

File: pkg/health/health.go [hunk 26]
```diff
@@ -0,0 +1,205 @@
...
+			podNotRunning = append(podNotRunning, podList.Items[i])
+		} else {
+			podRunning = append(podRunning, podList.Items[i])
+		}
+	}
+	return podRunning, podNotRunning
+}
+
+func checkPgStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string) {
+	_, PgStateEntryList := unMarshalCephStatus(context, operatorNamespace, clusterNamespace)
```
Comment (travisn) [inline pkg/health/health.go:162]:
```suggestion
	_, pgStateEntryList := unMarshalCephStatus(context, operatorNamespace, clusterNamespace)
```

Comment (subhamkrai) [reply pkg/health/health.go]: done

**Comment by subhamkrai** [comment]
Go ahead with this pr, since go-test ci is passing `with-pr-changes-in-custom-namespace` is failing but only valid for a bash script.
